spec.md

```diff
@@ -222,6 +222,29 @@ that contains both context and data).
   * OPTIONAL
 
 
+## Known extensions
+The following extensions are in active use. To better support promotion of
+extensions to full-features, Event Sources SHOULD use known extensions when
+they already suit the Event Source's needs.
+
+### jwt_claims
+* Type: Object
+* Description: If the Event Source supports JWT authentication, this extension
+  allows the Source to provide information about which user triggered the
+  occurrence for the Event. The jwt_claims extension is unsigned claims portion
+  of a JWT. This allows event systems to forward authentication information
+  without fear of leaking end-user authorization. An empty object MAY be used
```

Comment:
Yes, this is part of the reason why I had suggested passing the actual claims Object instead of the signed JWT token. It mitigates part of the risk to be PII disclosure rather than also allowing unauthorized impersonation of the user. Do you have wording you'd suggest?

Reply:
If the purpose of this field is to transmit information about the user who triggered the event then include only the `sub` claim is enough? There is nothing PII in `sub` claim and no unauthorised user can impersonate using just the `sub` claim. 

However, if the purpose of including this information is to actually authenticate the sender of the event, then you would need to include the full token. How is this any more secure than `HTTP Bearer Token` scheme? 

Going back to the wording question - my suggestion would be to remove any statement about how the spec helps stop leaking of sensitive information.

Reply:
There's a lot of good stuff densely packed in this comment. I'll try to inline my response:

> There is nothing PII in sub claim

A username is commonly considered PII, for example when the username is an email address or a common alias. Our architectures will be subjected to the same scrutiny internally regardless of whether we add more claims or not.

> and no unauthorised user can impersonate using just the sub claim.

No unauthorized actor can _impersonate_ the user unless they have the issuer's private key either. I thought about using just the sub claim. I backed out for two reasons:
1. When designing a platform, I often use the litmus test of "could this be used to itself build a platform?" All of these claims would be needed by a platform developer; their customers would each have their own issuer and that issuer would mint tokens for the platform developer's customers' customers.
2. These seem necessary for serverless/event-driven development to not regress from more traditional architectures (see next inlined comment)

> However, if the purpose of including this information is to actually authenticate the sender of the event, then you would need to include the full token. How is this any more secure than HTTP Bearer Token scheme?

This is a complex/emerging topic; some background might help. Our oldest best practices consider a single identity. In many systems (esp. with FaaS and events), there tend to be two kinds of identities:

1. Software identity (in Google, a "service account") that the developer manually configures and controls with access control like IAM. This identity system is what, for example, authorizes Google Cloud Storage to publish its events to the developer's Pub/Sub topic. 
2. End user identities that customers of a product use.

In traditional monolith architecture where the developer builds an API server for every client <-> server communication, this is easily accessed. The rise of BaaS means that a platform provider, not the developer, now handles the original request and is the authenticated identity that calls developer code (e.g. FaaS).

This information is not enough to further act _as_ the end user (though if the Action that receives the event has an "act as" permission it may be able to exchange the claims for a new token), but it is enough to act _on behalf_ of the user.

> my suggestion would be to remove any statement about how the spec helps stop leaking of sensitive information.

I completely agree. I'll remove this since it added confusion/contention.